### PR TITLE
[tune] Fix and enable SigOpt tests

### PR DESCRIPTION
**Pull request by krfricke**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by simon-mo** [comment]
FYI Travis won't inject secrets to PR environment

**Comment by richardliaw** [comment]
Looks like the sigopt tests were skipped

**Comment by krfricke** [comment]
Ah thanks, I remember reading that somewhere. There's a couple of things here to fix, I'll get to that tomorrow and disable skip tests when SIGOPT_KEY is not found in the environment. Unfortunately that means we won't see if the tests pass on travis until we merged. (The SigOpt tests specifically pass locally, though).

File: .travis.yml [hunk 1]
```diff
@@ -403,6 +403,8 @@ matrix:
         - TFP_VERSION=0.8
         - TORCH_VERSION=1.7
         - PYTHONWARNINGS=ignore
+        # SIGOPT_KEY
+        - secure: "XP07W/LnvHdtUtlp2PHCa/DuZJu3sPvShi+mIsadSwGn5PdDGOyPYwCHVzD7GC10fJJTwBwIRqFLptHtYQpGVCZgwNdmVlOu0uLEqvFU2GxofkuJeCM7WDSTSNCZb5eD+OdzIAd3vdNNXslqGzfKF4P8FQVNl3L/q6BaxRoWjPXTu+mAJs6iGswBlXacKn8tkTdYzZp8dYSsNTvQdqOG6+EwsD/orBvQOv/QlTXhyHi5gbApc5murYHvNny+oJAOIMsINnefED+e4WIK6dbMoaB9XLOCVcOFwMyIaLPl9A2dHl2iTLJCveMGzP9HpAvu/u02vDTxTRCO8C9s9fMgL8DBy4M+BfHPa2cl+ldtHoykKGZ8NHyhKoK3TwUxo3lmlTn44yBROo7IZOXtt7WZsfGUrnc91pNzDBSzyW1vsXdBQyo5LfMrWiiISHV3TLWUqHyK+/ADjJotmCvZ47FiJopGpyzT0jesprcQZ3cV2tzQdQxFfMKQS29fm0Aeot0W5YVD+K/EXsmQi/D073fCYOmXNB6owcGmAwsGsnaieHI3ymttaKp2u/g/HcXbhE/LydRS30hwX6qRl/+OJVrAiu5se5qmILDxYP7JXTRKfcUy3xuzrYaFW/m7DwSrPFCveCqJTZLBsXyGzGPRzNtxbtJEORRMOvJA7tm2RwbAdhE="
```
Comment (richardliaw) [inline .travis.yml:407]:
```suggestion
```

I just added the sigopt key as a private key in the travis UI, if you look at the tests it looks like it works :)

File: .travis.yml [hunk 2]
```diff
@@ -461,6 +461,8 @@ matrix:
...
```
Comment (richardliaw) [inline .travis.yml:465]:
```suggestion
```

**Comment by krfricke** [comment]
Ah awesome, thanks!

**Review by richardliaw — APPROVED** [review]
Looks good now!

**Comment by richardliaw** [comment]
(will wait one last time for tests to pass)

**Comment by krfricke** [comment]
The one failing test is unrelated to this PR - should be ready to merge!

**Pull request merged**